src/components/PaginationContainer.ts

```diff
@@ -0,0 +1,150 @@
+    }
+
+    componentWillUnmount() {
+        dojoConnect.disconnect(this.navigationHandler);
+    }
+
+    public static setMessageStatus(currentOffSet: number, offSet: number, maxPageSize: number): string {
+        let fromValue = currentOffSet + 1;
+        let toValue = 0;
+        if (maxPageSize === 0) {
+            fromValue = 0;
+        } else if (maxPageSize < offSet || (currentOffSet + offSet) > maxPageSize) {
+            toValue = maxPageSize;
+        } else {
+            toValue = currentOffSet + offSet;
+        }
+
+        return window.mx.ui.translate(
+            "mxui.lib.MxDataSource",
+            "status",
+            [ fromValue, toValue, maxPageSize ]);
+    }
+
+    private renderPageButton(): ReactElement<PaginationProps> | null {
+        if (this.state.validationPassed) {
+            return createElement(Pagination, {
+                maxPageSize: this.state.maxPageSize,
+                offSet: this.state.offSet,
+                onClickAction: this.updateListView,
+                setMessageStatus: PaginationContainer.setMessageStatus,
+                showPageButton: this.state.showPageButton
+            });
+        }
+
+        return null;
+    }
+
+    private validateConfigs() {
+        if (!this.state.validationPassed) {
+            const queryNode = findDOMNode(this) as HTMLElement;
+            const targetNode = ValidateConfigs.findTargetNode(queryNode);
+            let targetListView: ListView | null = null;
+
+            if (targetNode) {
+                targetListView = dijitRegistry.byNode(targetNode);
+                this.setState({ targetNode });
+
+                if (targetListView) {
+                    this.transformListView(targetNode, targetListView);
+                    this.setState({ targetListView });
+                }
+            }
+            const validateMessage = ValidateConfigs.validate({
+                ...this.props as WrapperProps,
+                queryNode: this.state.targetNode,
+                targetListview: this.state.targetListView,
+                validate: !this.state.findingListviewWidget
+            });
+            this.setState({ findingListviewWidget: false, validationPassed: !validateMessage });
+        }
+    }
+
+    private transformListView(targetNode: HTMLElement, listView: ListView) {
+        const buttonNode = targetNode.querySelector(".mx-listview-loadMore") as HTMLButtonElement;
+
+        this.setState({
+            maxPageSize: listView._datasource._setSize,
+            offSet: listView._datasource._pageSize,
+            showPageButton: !((listView._datasource._pageSize >= listView._datasource._setSize) && this.props.hideUnusedPaging)
+        });
+
+        if (buttonNode) {
+            buttonNode.style.display = "none";
+        }
+
+        this.listListViewHeight = targetNode.clientHeight;
+    }
+
+    private updateListView(offSet: number) {
+        const { targetListView, targetNode, validationPassed } = this.state;
```

Comment:
Unnecessary destructuring. Inline.

Reply:
Its necessary because in the next line the validation done against the states in the if statement is longer than the accepted number of lines